baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/ForbidJavaxParameterTypeTest.java

```diff
@@ -0,0 +1,131 @@
+    }
+
+    @Test
+    void failsIfIncorrectTypeIsSuppliedViaParameter() {
+
+        helper().addSourceLines(
+                        "Test.java",
+                        "package test;",
+                        "import javax.ws.rs.Path;",
+                        "import javax.ws.rs.GET;",
+                        "import com.palantir.errorprone.ForbidJavax;",
+                        "class Test {",
+                        "  public void requiresJakarta(@ForbidJavax Object o) {}",
+                        "  ",
+                        "  public static class Inner {",
+                        "    @Path(\"foo\")",
+                        "    @GET",
+                        "    public String doGet() { return \"hi\"; }",
+                        "  }",
+                        "  public static void main(String[] args) {",
+                        "    Inner inner = new Inner();",
+                        "    // BUG: Diagnostic contains: ForbidJavaxParameterType",
+                        "    new Test().requiresJakarta(inner);",
+                        "  }",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    void failsIfIncorrectTypeIsSuppliedViaMethodReturn() {
+
+        helper().addSourceLines(
+                        "Test.java",
+                        "package test;",
+                        "import javax.ws.rs.Path;",
+                        "import javax.ws.rs.GET;",
+                        "import com.palantir.errorprone.ForbidJavax;",
+                        "class Test {",
+                        "  public void requiresJakarta(@ForbidJavax Object o) {}",
+                        "  ",
+                        "  public static class Inner {",
+                        "    @Path(\"foo\")",
+                        "    @GET",
+                        "    public String doGet() { return \"hi\"; }",
+                        "  }",
+                        "  public static Inner getInner() { return new Inner(); }",
+                        "  public static void main(String[] args) {",
+                        "    // BUG: Diagnostic contains: ForbidJavaxParameterType",
+                        "    new Test().requiresJakarta(getInner());",
+                        "  }",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    @Disabled("Unfortunately disabled because we can't have both annotation types on the classpath")
+    void succeedsIfGivenJakartaType() {
+
+        helper().addSourceLines(
+                        "Test.java",
+                        "package test;",
+                        "import jakarta.ws.rs.Path;",
+                        "import jakarta.ws.rs.GET;",
+                        "import com.palantir.errorprone.ForbidJavax;",
+                        "class Test {",
+                        "  public void requiresJakarta(@ForbidJavax Object o) {}",
+                        "  ",
+                        "  public static class Inner {",
```

Comment:
Concretely, I think if you move the `Inner` definition to a nested class within this test class, the helper should still be able to reference it, but I don't think it would be detected due to the ClassTree search

Reply:
Just added a test where it's moved to a separate file...seems to work? Or is there another case I'm not thinking of?

Reply:
I've created a PR with an example that fails: https://github.com/palantir/gradle-baseline/pull/2367

(accidentally replied to the wrong thread a moment ago, fixed)